### PR TITLE
ci: migrate rebase-pr CI script to the circleci directory

### PR DESCRIPTION
**Pull request by josephperrott**
Migrate the rebase-pr script used on CI out of tools and into the circleci
directory.  Additionally removes its support for running in the local
repository as this is now better handled by `ng-dev pr rebase <pr-number>`.

### CODE REVIEW

File: .circleci/rebase-pr.js [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+    commonAncestorSha,
+  };
+}
+
+
+_main().catch(err => {
+  console.log('Failed to rebase on top of target branch.\n');
+  console.error(err);
+  process.exitCode = 1;
+});
```
Comment (gkalpak) [inline .circleci/rebase-pr.js:209]: Why move this to the bottom? Having it above helpers (as it was before) is more readable imo (but maybe that's just my personal preference).

File: .circleci/rebase-pr.js [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+  try {
+    output += execSync_(command, {stdio: ['pipe', 'pipe', 'pipe']}).toString().trim();
+  } catch (err) {
+    allowStderr && console.error(err.stderr.toString());
+    output.code = err.status;
+  }
+  return output;
+}
+
+// Run
```
Comment (gkalpak) [inline .circleci/rebase-pr.js:45]: The idea was to have this comment along with the code that kicks off the work (so this has to stay with the `_main()` call :grin:).

File: tools/rebase-pr.js [hunk 3]
```diff
@@ -1,102 +0,0 @@
...
-
-/**
- * **Usage:**
- * ```
- * node rebase-pr <github-repository> <pull-request-number>
- * ```
- * **Example:**
- * ```
- * node rebase-pr angular/angular 123
- * ```
```
Comment (gkalpak) [inline tools/rebase-pr.js:17]: This comment seems to not have been retained in the new script. Any particular reason? (It still seems relevant and useful.)

File: .circleci/rebase-pr.js [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+ * Since a GitHub PR is not necessarily up to date with its target branch, it is useful to rebase
+ * prior to testing it on CI to ensure more up to date test results.
+ *
+ * **NOTE:**
+ * This script cannot use external dependencies or be compiled because it needs to run before the
+ * environment is setup.
+ * Use only features supported by the NodeJS versions used in the environment.
+ */
+// tslint:disable:no-console
+const {execSync: execSync_} = require('child_process');
```
Comment (gkalpak) [inline .circleci/rebase-pr.js:22]: This renaming seems redundant. Why not just:

```suggestion
const {execSync} = require('child_process');
```

**Review by gkalpak — APPROVED** [review]
A couple of minor comments/questions. Otherwise lgtm :+1:
Also, consider adding a comment on why we don't use the `ng-dev` command on CI.

**Comment by AndrewKushnir** [comment]
FYI, adding "cleanup" label based on comments from @gkalpak. Thank you.

File: tools/rebase-pr.js (hunk 3 above)
Comment (josephperrott) [reply tools/rebase-pr.js:17]: The command not longer obtains this information from the command line arguments, instead it gathers the information it needs from environment variables.

Comment (gkalpak) [reply tools/rebase-pr.js:17]: Oh, right, I missed that 😅.

**Pull request closed without merging**